### ISSUE TITLE
Enable select as default label value

### ISSUE THREAD

**Issue by abergmeier-dsfishlabs** [bazelbuild/bazel#2026]
### Description of the problem / feature request / question:

`select` is not enabled as Label default

### If possible, provide a minimal example to reproduce the problem:

    foo_repository = repository_rule(
      implementation = _impl,
      attrs = {
        "binary": attr.label(
          default = select({
            "//:config_linux_x86_64": Label("@foo//:binary"),
            "//:config_win_x86_64": Label("@foo_win//:binary"),
           }),

### Environment info
* Bazel version (output of `bazel info release`): 0.3.2

###  Have you found anything relevant by searching the web? (e.g. GitHub issues, email threads in the [bazel-discuss@googlegroups.com](https://groups.google.com/forum/#!forum/bazel-discuss) archive)
Nope.

### Anything else, information or logs or outputs that would be helpful?
The error message is:

    expected Label or Label-returning function or NoneType for 'default' while calling label but got select of Label instead

**Comment by petemounce** [comment]
Aside: @meteorcloudy @laszlocsomor - this looks like it was done via a template. Github has a feature where if one creates (for example) `.github/ISSUE_TEMPLATE.md` in the repo, then github will use that to pre-fill the issue description when someone wants to open one. Learned from [homebrew-core](https://github.com/Homebrew/homebrew-core/blob/master/.github/ISSUE_TEMPLATE.md). It works for pull requests too.

**Comment by laszlocsomor** [comment]
@petemounce : Yes, we just started using that GH feature :) See https://github.com/bazelbuild/bazel/commit/ca2c184affbc6da871b8c71d2dd3d54a2ef5f8d9

**Comment by petemounce** [comment]
Ah :)

**Comment by laurentlb** [comment]
The only workaround for now is to create a macro, that calls your rule with the appropriate default value.

**Comment by abergmeier-dsfishlabs** [comment]
Adding a macro indirection I get

    select() cannot be used in macros called from WORKSPACE files

So changing to using `repository_ctx.os` directly.

**Comment by abergmeier-dsfishlabs** [comment]
@damienmg Since you cannot depend on non-source labels, you can only reference prebuilt generators for `repository_rule`. E.g.:

```
my_repo = repository_rule(
    implementation = _my_repo_impl,
    attrs = {
        "_generator": attr.label(default = Label("@root//repo_gen:repo_gen_prebuilt"), executable = True, cfg="host"),
```

Here `repo_gen_prebuilt` generates a file and package structure appeasing to Bazel.
Now, for every platform you need a different `_generator` binary of course. How do you do that elegantly without `select`?

**Comment by damienmg** [comment]
Use rctx.os.name? If you use the Label function it should just work,
